src/main/java/org/cactoos/scalar/Equals.java

```diff
@@ -49,14 +50,14 @@
     /**
      * The second scalar.
      */
-    private final Scalar<? extends T> second;
+    private final Scalar<? extends R> second;
 
     /**
      * Ctor.
      * @param source The first scalar to compare.
      * @param compared The second scalar to compare.
      */
-    public Equals(final T source, final T compared) {
+    public Equals(final T source, final R compared) {
```

Comment:
@victornoel I'm curious. Why the two values to compared are not the same type here ? What do you want to resolve by doing that like this way ?

Reply:
@baudoliver7 I'm not 100% sure it's the best solution, but according to tests it is the only one I managed to make work to solve #1174. Basically, it's enough that the compared is of the same type (or subtype) as the type parameter of `Comparable` (which `T` implements), it doesn't need to be of type `T` itself.

Reply:
@victornoel Thanks, It feels good to have my ideas fixed.